log/logger.go

```diff
@@ -0,0 +1,52 @@
+package log
+
+import (
+	"log/slog"
+	"os"
+)
+
+// a structured logger encapsulating log/slog
+var (
+	logger   *slog.Logger
+	logLevel = os.Getenv("LOG_LEVEL")
+)
+
+// init logger
+func init() {
+	var opts *slog.HandlerOptions
+	switch logLevel {
+	case "ERROR":
+		opts = &slog.HandlerOptions{Level: slog.LevelError}
+	case "WARN":
+		opts = &slog.HandlerOptions{Level: slog.LevelWarn}
+	case "DEBUG":
+		opts = &slog.HandlerOptions{Level: slog.LevelDebug}
+	default:
+		opts = &slog.HandlerOptions{Level: slog.LevelInfo}
+	}
+	logger = slog.New(slog.NewTextHandler(os.Stdout, opts))
+}
+
+// encapsulating func in log/slog
+// args: expected in key:value pairs
+func Info(msg string, args ...any) {
+	logger.Info(msg, args...)
+}
+
+// encapsulating func in log/slog
+// args: expected in key:value pairs
+func Debug(msg string, args ...any) {
+	logger.Debug(msg, args...)
+}
+
+// encapsulating func in log/slog
+// args: expected in key:value pairs
+func Warn(msg string, args ...any) {
+	logger.Warn(msg, args...)
+}
+
+// encapsulating func in log/slog
+// args: expected in key:value pairs
+func Error(msg string, args ...any) {
+	logger.Error(msg, args...)
+}
```

Comment:
Should these be methods on the logger struct?

e.g.:

```go
func (logger *slog.Logger) Error(msg string, args ...any) {
    logger.Error(msg, args...)
}
```

Reply:
@CallumNZ I think the usage of `log` or `logger` is not via methods, we'll just use it as an exported function from the package. 

Take an example,  we invoke `log.Println()` in place,  without `mylogger := log.NewLogger()` then call `mylogger.Error()`.

Reply:
`Take an example, we invoke log.Println() in place, without mylogger := log.NewLogger() then call mylogger.Error().`

I don't understand.

Reply:
NVM I read it wrong. LOL

Reply:
Example of this approach:

https://github.com/GeoNet/field-monitor-project/blob/main/internal/fmp_logger/fmp_logger.go

Then we simply call the package's method without initializing
https://github.com/GeoNet/field-monitor-project/blob/main/cmd/fmp-polling/main.go#L84